### PR TITLE
lua: "nvim -l" scriptname in _G.arg[0], eliminate bump-deps.sh

### PR DESCRIPTION
**Pull request by justinmk**
- store "nvim -l" scriptname in `_G.arg[0]`
- Since https://github.com/neovim/neovim/pull/18706 we can use `nvim -l` to run Lua, so we don't need to wrap things in shell scripts nor use `:luafile require(..)`.

@clason I tested a few things but would be great if you can confirm.

### CODE REVIEW

File: scripts/bump_deps.lua [hunk 1]
```diff
@@ -1,343 +1,449 @@
...
 --    # create branch
 --    nvim -es +"lua require('scripts.bump_deps').create_branch()"
 
 local M = {}
 
 local _trace = false
-local required_branch_prefix = "bump-"
-local commit_prefix = "build(deps): "
+local required_branch_prefix = 'bump-'
+local commit_prefix = 'build(deps): '
```
Comment (justinmk) [inline scripts/bump_deps.lua:9]: stylua

File: scripts/bump_deps.lua [hunk 2]
```diff
@@ -1,343 +1,449 @@
...
+      args.commit = _G.arg[i+1]
+    elseif _G.arg[i] == '--head' then
+      args.head = true
+    end
+  end
+  return args
+end
+
+local scriptname = vim.v.argv[#vim.v.argv - #_G.arg]
+local is_main = scriptname:match('bump_deps.lua')
```
Comment (justinmk) [inline scripts/bump_deps.lua:425]: Need some sort of `vim.is_main()` (?) thing that scripts can use to decide whether to run like a script or like a module. Similar to python `__name__ == "__main__"` pattern.

File: scripts/bump_deps.lua [hunk 3]
```diff
@@ -1,343 +1,449 @@
...
+
+if is_main then
+  local args = parseargs()
```
Comment (justinmk) [inline scripts/bump_deps.lua:428]: do we need `vim.argparse()` ? perhaps using https://github.com/mpeterv/argparse

File: scripts/bump_deps.lua [hunk 4]
```diff
@@ -1,343 +1,446 @@
...
+      args.commit = _G.arg[i+1]
+    elseif _G.arg[i] == '--head' then
+      args.head = true
+    end
+  end
+  return args
 end
 
-return M
+local is_main = _G.arg[0]:match('bump_deps.lua')
```
Comment (justinmk) [inline scripts/bump_deps.lua:422]: This is not too bad but not 100% reliable. Might still want some sort of `vim.is_main()` (?) thing that scripts can use to decide whether to run like a script or like a module. Similar to python `__name__ == "__main__"` pattern.

Comment (ii14) [reply scripts/bump_deps.lua:422]: Maybe something roughly like this? When neovim runs the script:
```lua
local main = loadfile('...')
_G.arg = { ... } -- set args from command line
_G.__main__ = main -- set __main__ to the main function
main()
```
And then inside the script:
```lua
if __main__ == debug.getinfo(1, 'f').func then
  ...
end
```

Comment (justinmk) [reply scripts/bump_deps.lua:422]: that looks promising. but where/what is the `main()` function?

Comment (ii14) [reply scripts/bump_deps.lua:422]: `main` is the function that is loaded from `-l <file>` with `loadfile`. Right now `nlua_exec_file` does `loadfile` and then immediately executes it. It'd just need to also set that function as some variable (but only in this case obviously)

Comment (justinmk) [reply scripts/bump_deps.lua:422]: cool idea. should `debug` be avoided though, for such common functionality? 

wonder if we can use the C API to set a magic local inside the loadfile() closure.

**Pull request merged**